tracing-subscriber/src/fmt/fmt_subscriber.rs

```diff
@@ -1221,20 +1212,87 @@ where
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 struct Timings {
-    idle: u64,
-    busy: u64,
+    idle: Duration,
+    busy: Duration,
     last: Instant,
+    entered_count: u64,
 }
 
+/// Tracks the idle and busy times of a span.
+///
+/// The number of times the span has been entered is tracked to determine the current state. This
+/// ensures that entering and exiting the span multiple times out of order does not affect the
+/// timings.
```

Comment:
This should also be on the struct itself, not the impl block.

Or are docs for impl blocks also shown on the structs?

Reply:
You're right these belong on the struct.
Fixed in [a046e31](https://github.com/tokio-rs/tracing/pull/3038/commits/a046e31e23e5c1d4c4e854360a17b6301221c373)